core/services/workflows/cmd/cre/utils/runner.go

```diff
@@ -118,37 +129,37 @@ func NewRunner(hooks *RunnerHooks) *Runner {
 }
 
 // run instantiates the engine, starts it and blocks until the context is canceled.
-func (r *Runner) run(
+func (r *Runner) Run(
 	ctx context.Context,
 	binary, config []byte,
 	cfg RunnerConfig,
 ) {
-	cfg.lggr.Infof("executing engine in process: %d", os.Getpid())
+	cfg.Lggr.Infof("executing engine in process: %d", os.Getpid())
 
 	registry, services := r.hooks.Initialize(ctx, cfg)
 
 	billingAddress := ""
-	if cfg.enableBilling {
+	if cfg.EnableBilling {
 		billingAddress = "localhost:4319"
 	}
 
-	engine, err := NewStandaloneEngine(ctx, cfg.lggr, registry, binary, config, billingAddress, v2.LifecycleHooks{})
+	engine, triggerSub, err := NewStandaloneEngine(ctx, cfg.Lggr, registry, binary, config, billingAddress, cfg.LifecycleHooks)
 	if err != nil {
 		fmt.Printf("Failed to create engine: %v\n", err)
 		os.Exit(1)
 	}
 
 	services = append(services, engine)
 
-	r.hooks.BeforeRun(ctx, cfg, registry, services)
+	r.hooks.BeforeRun(ctx, cfg, registry, services, triggerSub)
 
 	err = engine.Start(ctx)
 	if err != nil {
 		fmt.Printf("Failed to start engine: %v\n", err)
 		os.Exit(1)
 	}
 
-	<-ctx.Done()
+	r.hooks.Close(ctx, cfg, registry, services)
```

Comment:
why move this blocking call into the hook?  if someone overwrites the close method without a blocking call this will break the run

Reply:
yeah I know but I need to be able to override it to something like:
```
closeFn := func(context.Context, simulator.RunnerConfig, *capabilities.Registry, []services.Service) {
		<-initializedCh

		// Manual trigger cron
		baseLggr.Infow("Running trigger", "trigger", triggerToRun.GetId())
		triggerCaps.ManualCronTrigger.ManualTrigger(ctx, time.Now())

		select {
		case <-executionFinishedCh:
			baseLggr.Debugw("Execution finished signal received")
		case <-time.After(15 * time.Second):
			baseLggr.Debugw("Timeout waiting for execution to finish")
		}
	}
```

so it blocks on waiting for initialization, triggers the manual trigger, then waits for execution finished